packages/server/src/db/index.js

```diff
@@ -352,6 +352,168 @@ async function getNewGrantsForAgency(agency) {
     return rows;
 }
 
+async function buildPaginationParams(args) {
+    const { currentPage, perPage } = args;
+    let { isLengthAware } = args;
+
+    if (!currentPage) {
+        throw Error('Missing current Page');
+    }
+
+    if (!perPage) {
+        throw Error('Missing per page');
+    }
+
+    if (isLengthAware === undefined || isLengthAware === null) {
+        isLengthAware = true;
+    }
+
+    return { currentPage, perPage, isLengthAware };
+}
+
+/*
+    filters: {
+        reviewStatuses: List[Enum['Interested', 'Result', 'Rejected']],
+        eligibilityCodes: List[String],
+        includeKeywords: List[String],
+        excludeKeywords: List[String],
+        opportunityNumber: String,
+        fundingTypes: List[Enum['CA, 'G', 'PC' ,'O']]
+        opportunityStatuses: List[Enum['posted', 'forecasted', 'closed']],
+        opportunityCategories: List[Enum['Other', 'Discretionary', 'Mandatory', 'Continuation']],
+        costSharing: Enum['Yes', 'No'],
+        agencyCode: String,
+        postedWithinDays: number,
+    },
```

Comment:
@joshgarza I added the type definition of the filters here. As part of your validation, it would be great to get the data in this structure prior to saving a search as well as when requesting grants.